### ISSUE TITLE
Color contrast & accessibility

### ISSUE THREAD

**Issue by mknadler** [18mr/18mr#37]
TL;DR: The color contrast is too low in a bunch of places (including but not limited to the pagination widget, bylines, and links), and it'd be great if they were made higher in those places for accessibility reasons. It'd also be great if links were underlined or italicized, for the same reasons. 

---

The long-winded, semi-exhaustive version:

The `Web Accessibility Initiative` (WAI) published `Web Content and Accessibility Guidelines` (WCAG); among (many) other things, the WCAG guidelines [include specific levels of relative luminance contrast](https://www.w3.org/WAI/WCAG20/quickref/#visual-audio-contrast-contrast) required for the accessibility of text. In other words, they tell the internet how to make it so that just about everyone can read the internet.

This isn't actually strictly a _requirement_ for 18mr, because 18mr is not ~as far as I know~ a government project. However, I'd recommend these changes for the sole reason that there are probably some people out there who are trying to read 18mr materials and are having a tricky time of it.

I think this is whole color contrast standards deal is a really neat concept -- this group looked at how eyes work, and how vision deficiencies work, and determined that the way our eyes can 'go bad' are so myriad that it's impossible to say something like "yeah, blue on gray (or some other color combo) is always readable" and be right even most of the time. Instead, they used ⭐️ formulas and science 🌟 to determine that if the relative luminance of the lighter of the two colors is measurably different enough from the relative luminance of the darker of the two colors, it will be legible to the vast majority of people.

The AA (think 'two stars out of three') requirement, by the way, is:
- 4.5:1 ratio for normal text
- 3:1 ratio for bold or large text

If you're wondering "cool, but how do I measure this?": [this tool](http://webaim.org/resources/contrastchecker/) is helpful.

For the most part, https://18millionrising.org meets or exceeds these standards, which rules. For instance, the main body text / main background combination hits at a 13.76:1 contrast ratio. I did, however, find a few spots that might be difficult to read for people with low vision or other conditions that can affect reading ability (whether that be colorblindness, age-related eye problems, or simply a dimmed monitor).

---
## Example: Pagination

![image](https://user-images.githubusercontent.com/3596891/28810056-8f8f9366-7655-11e7-866f-4f62e5a0e414.png)
The leftmost number is the current page; the foreground color is #e8e8e8, and the background color is #8bc63e. The ratio here is 1.67:1, which is lower than the limit for large/bold text.
The middle number is a page other than the one the person using the site is currently on; the foreground color is #ebdd14, and the background color is #2b99c1. The ratio here is 2.32:1, which is also lower than the limit, but a bit better.
The rightmost number is the hover state of a non-current page; the foreground color changes on hover to #221e20, and the background color remains #2b99c1. The ratio here is a solid 5.04:1, which roundly passes.

My suggestion here: revisit the colors here; it seems likely that other colors from the palette could be used in various combination here and be both enticing and legible.

---
## Example: Byline

![image](https://user-images.githubusercontent.com/3596891/28810825-ce819552-7659-11e7-88f8-9d737496c919.png)

The site background color is #f6eeda; the text color for the byline is #828282. The contrast ratio here is 3.32:1, which would be fine if the text were larger or bolder.

This is also the case for the view from inside of the post:
![image](https://user-images.githubusercontent.com/3596891/28810978-9895c6a6-765a-11e7-86d0-0a524b1b3306.png)

My suggestion here: either make the byline color a hair darker, or make the bylines larger or bolder both inside and outside of posts.

--
## Example: Links

![image](https://user-images.githubusercontent.com/3596891/28811082-05ad921e-765b-11e7-8ab1-66c3f405f6e6.png)

The pink is #ea2485; the background color is #f6eeda; the surrounding dark gray text is #222222.

The pink:background ratio is 3.57:1, the pink:gray ratio is 3.86:1, and the the gray:background ratio is still 13.76:1.

The WCAG guidelines require something _else_ that I haven't even mentioned yet: [color should not be used as the only visual means of conveying information](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-without-color). [Technique G183](https://www.w3.org/TR/2008/NOTE-WCAG20-TECHS-20081211/G183) explains that, when color _is_ being used as the sole means of conveying that something is a link (or something like a link), it should have a 3:1 ratio with the surrounding text. Which, currently, it does!

The problem is that the link:background ratio fails at this small size (and would fail the AAA standard if it were bolder).

What this means all together is that while normally-sighted people looking at 18mr might be seeing this: 
![image](https://user-images.githubusercontent.com/3596891/28811581-41dc0ec0-765e-11e7-910e-fdc3d8316e5a.png)

other people might be seeing something more like this:
![image](https://user-images.githubusercontent.com/3596891/28811637-899b767e-765e-11e7-804d-1c0fc4aeaa1c.png)

My suggestion here is twofold: 
1. Either lighten the background color a bit, or change the pink so that it hits a higher relative luminance ratio against the existing background;
2. Add underlines or italicization of links

**Comment by catachresis** [comment]
This is super useful, and these changes are important to implement.

In pagination, I think we could do with different colors; I can pull some colors if that's helpful.

I have a slight preference for italicization of links, and I think lightening the background is probably the best solution here.

**Comment by debracleaver** [comment]
I want to chime in with an enthusiastic plus one on all of this.  Also, Miriam, i'm sure you're as obsessed with this tool as I am: http://webaim.org/resources/contrastchecker/
